indico/modules/receipts/util.py

```diff
@@ -246,7 +246,9 @@ def get_dummy_preview_data(custom_fields: dict) -> dict:
     try:
         return TemplateDataSchema().load({
             **dummy_data,
-            'custom_fields': {f['name']: _get_default_value(f) for f in custom_fields},
+            'custom_fields': {f['name']: (custom_fields_values[f['name']]
```

Comment:
`custom_fields_values.get(f['name'], _get_default_value(f))` instead of that strange check if the key exists.

And that other check can be avoided by doing this near the beginning of the function:

```python
if custom_fields_values is None:
    custom_fields_values = {}
```